Source/WebCore/platform/SharedBuffer.cpp

```diff
@@ -67,6 +74,38 @@ Ref<FragmentedSharedBuffer> FragmentedSharedBuffer::create(DataSegment::Provider
     return adoptRef(*new FragmentedSharedBuffer(WTFMove(provider)));
 }
 
+#if PLATFORM(COCOA)
+std::optional<Ref<FragmentedSharedBuffer>> FragmentedSharedBuffer::fromIPCData(IPCData&& ipcData)
+{
+    return WTF::switchOn(WTFMove(ipcData), [](Vector<std::span<const uint8_t>>&& data) -> std::optional<Ref<FragmentedSharedBuffer>> {
+        if (data.empty())
+            return SharedBuffer::create();
+
+        size_t size = 0;
+        for (auto& span : data)
+            size += span.size();
+
+        if (useUnixDomainSockets || size < minimumPageSize) {
+            if (data.size() == 1)
+                return SharedBuffer::create(data[0]);
+            Ref sharedMemoryBuffer = FragmentedSharedBuffer::create();
+            for (auto& span : data)
```

Comment:
I think `auto` is fine for span, no need for `auto&`. spans are super cheap to copy.